# HG changeset patch
# User Nils Bruin <nbruin@sfu.ca>
# Date 1299001283 28800
# Node ID 3f933d7710d60ce9576589d28a9772c65fbbb596
# Parent  ab67d83570dd001f2290e75730689b658ec4a75d
Lazy import of maxima in calculus using sage.misc.lazy_import

diff -r ab67d83570dd -r 3f933d7710d6 sage/calculus/calculus.py
--- sage/calculus/calculus.py	Tue Mar 01 08:26:37 2011 -0800
+++ sage/calculus/calculus.py	Tue Mar 01 09:41:23 2011 -0800
@@ -383,8 +383,8 @@
     sage: sage.calculus.calculus.maxima('f1')
     f1
 """
-import sage.interfaces.maxima_lib
-maxima = sage.interfaces.maxima_lib.maxima_lib
+from sage.misc.lazy_import import lazy_import
+lazy_import('sage.interfaces.maxima_lib','maxima')
 # This is not the same instance of Maxima as the general purpose one
 #from sage.interfaces.maxima import Maxima
 #maxima = Maxima(init_code = ['display2d : false', 'domain : complex',
